sharedbuffers/mapped_struct.py

```diff
@@ -2461,6 +2874,61 @@ def map_file(cls, fileobj, offset = 0, size = None):
         rv._file = fileobj
         return rv
 
+    @classmethod
+    @cython.locals(
+        basepos = cython.ulonglong, curpos = cython.ulonglong, endpos = cython.ulonglong, finalpos = cython.ulonglong,
+        discard_duplicates = cython.bint, discard_duplicate_keys = cython.bint, index_elements = cython.size_t,
+        mapper = 'NumericIdMapper')
+    def merge(cls, parts, destfile = None, tempdir = None,
+            discard_duplicates = False, discard_duplicate_keys = False):
+        if destfile is None:
+            destfile = tempfile.NamedTemporaryFile(dir = tempdir)
+
+        dtype = cls.dtype
+        basepos = destfile.tell()
+
+        # Reserve space for the header
+        write = destfile.write
+        write(cls._Header.pack(0, 0))
+
+        indexpos = basepos + cls._Header.size
+
+        # Merge the indexes
+        def merge_all(parts):
+            if len(parts) == 1:
+                return parts[0]
+            else:
+                nparts = []
+                for i in xrange(0, len(parts), 2):
+                    if i+1 < len(parts):
```

Comment:
I this it'll be more readabot to check do :
```
if len(parts)%2 != 0:
     nparts.append(parts[-1])
```
after the loop. you'll also improve performance by doing just one "if"